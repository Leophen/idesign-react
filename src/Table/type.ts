export interface ColumnType {
  /**
   * 唯一标识
   */
  key: string;
  /**
   * 标题
   */
  title: string;
  /**
   * 宽度
   */
  width?: React.CSSProperties["width"];
  /**
   * 对齐方式
   */
  align?: React.CSSProperties["textAlign"];
}

export interface TableProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 表头配置数据
   * @default []
   */
  columns?: ColumnType[];
  /**
   * 表格数据源
   * @default []
   */
  data?: { [x: string]: any }[];
  /**
   * 表格列表最大高度，超出显示滚动条
   */
  maxHeight?: React.CSSProperties["height"];
  /**
   * 是否显示斑马纹
   * @default false
   */
  stripe?: boolean;
}

export interface ColumnGroupProps {
  /**
   * 列数据项
   * @default []
   */
  columns?: ColumnType[];
}

export interface TableHeadProps {
  /**
   * 列数据
   * @default []
   */
  columns?: ColumnType[];
}

export interface TableBodyProps {
  /**
   * 数据源
   * @default []
   */
  data?: { [x: string]: any }[];
  /**
   * 列数据
   * @default []
   */
  columns?: ColumnType[];
  /**
   * 表格列表高度，超出显示滚动条
   */
  maxHeight?: React.CSSProperties["height"];
  /**
   * 是否显示斑马纹
   * @default false
   */
  stripe?: boolean;
}
