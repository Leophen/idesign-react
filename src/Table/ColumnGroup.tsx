import React, { useMemo } from 'react';
import './index.scss';
import { ColumnGroupProps } from './type';

const ColumnGroup: React.FC<ColumnGroupProps> = (props) => {
  const {
    columns = []
  } = props

  const columnWidths = columns.map((ele) => ele.width).join("-");

  const cols = useMemo(() => {
    const cols: React.ReactElement[] = [];
    let mustInsert = false;
    for (let i = columns.length; i >= 0; i--) {
      const width = columns[i] && columns[i].width;
      if (width || mustInsert) {
        cols.unshift(
          <col
            key={i}
            width={width}
            style={{ width, minWidth: width, textAlign: columns[i].align }}
          />
        );
        mustInsert = true;
      }
    }
    return cols;
  }, [columnWidths]);

  return <colgroup>{cols}</colgroup>;
};

export default ColumnGroup;
