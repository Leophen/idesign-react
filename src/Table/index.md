---
nav:
  title: 组件
  path: /components
group:
  title: 表单组件
  order: 3
order: 11
---

# Table 表格

用于数据收集展示、分析整理、操作处理的表格。

## 基本用法

通过 `columns` 属性指定列配置，通过 `data` 属性指定数据源：

```tsx
import React from 'react';
import { Table } from 'idesign-react';

const App = () => {
  const columns = [
    {
      title: 'ID',
      key: 'id'
    },
    {
      title: 'Name',
      key: 'name'
    },
    {
      title: 'Age',
      key: 'age'
    },
    {
      title: 'Salary',
      key: 'salary'
    },
    {
      title: 'Address',
      key: 'address'
    },
    {
      title: 'Email',
      key: 'email'
    }
  ];

  const data = [
    {
      key: '1',
      id: 103761657,
      name: 'Jane Doe',
      age: 21,
      salary: 23000,
      address: '11 Park Road, Guangzhou',
      email: 'jane.doe@idesign.com'
    },
    {
      key: '2',
      id: 761651037,
      name: 'Alisa Ross',
      age: 22,
      salary: 25000,
      address: '22 Park Road, Guangzhou',
      email: 'alisa.ross@idesign.com'
    },
    {
      key: '3',
      id: 373567616,
      name: 'Kevin Sandra',
      age: 23,
      salary: 22000,
      address: '33 Park Road, Guangzhou',
      email: 'kevin.sandra@idesign.com'
    },
    {
      key: '4',
      id: 735646761,
      name: 'Ed Hellen',
      age: 24,
      salary: 17000,
      address: '44 Park Road, Guangzhou',
      email: 'ed.hellen@idesign.com'
    },
    {
      key: '5',
      id: 647616735,
      name: 'William Smith',
      age: 25,
      salary: 27000,
      address: '55 Park Road, Guangzhou',
      email: 'william.smith@idesign.com'
    },
    {
      key: '6',
      id: 616736475,
      name: 'Tim Cook',
      age: 26,
      salary: 36000,
      address: '66 Park Road, Guangzhou',
      email: 'tim.cook@idesign.com'
    },
    {
      key: '7',
      id: 475761636,
      name: 'Bill Gates',
      age: 27,
      salary: 12000,
      address: '77 Park Road, Guangzhou',
      email: 'bill.gates@idesign.com'
    },
    {
      key: '8',
      id: 364616775,
      name: 'Steve Jobs',
      age: 28,
      salary: 66000,
      address: '88 Park Road, Guangzhou',
      email: 'steve.jobs@idesign.com'
    }
  ];

  return (
    <Table columns={columns} data={data} />
  );
};

export default App;
```

## 自定义列宽

通过 `columns` 属性每一项的 `width` 来指定列宽，建议用百分比：

```tsx
import React from 'react';
import { Table } from 'idesign-react';

const App = () => {
  const columns = [
    {
      title: 'ID',
      key: 'id',
      width: '10%'
    },
    {
      title: 'Name',
      key: 'name',
      width: '10%'
    },
    {
      title: 'Age',
      key: 'age',
      width: '10%'
    },
    {
      title: 'Salary',
      key: 'salary',
      width: '10%'
    },
    {
      title: 'Address',
      key: 'address',
      width: '30%'
    },
    {
      title: 'Email',
      key: 'email',
      width: '30%'
    }
  ];

  const data = [
    {
      key: '1',
      id: 103761657,
      name: 'Jane Doe',
      age: 21,
      salary: 23000,
      address: '11 Park Road, Guangzhou',
      email: 'jane.doe@idesign.com'
    },
    {
      key: '2',
      id: 761651037,
      name: 'Alisa Ross',
      age: 22,
      salary: 25000,
      address: '22 Park Road, Guangzhou',
      email: 'alisa.ross@idesign.com'
    },
    {
      key: '3',
      id: 373567616,
      name: 'Kevin Sandra',
      age: 23,
      salary: 22000,
      address: '33 Park Road, Guangzhou',
      email: 'kevin.sandra@idesign.com'
    },
    {
      key: '4',
      id: 735646761,
      name: 'Ed Hellen',
      age: 24,
      salary: 17000,
      address: '44 Park Road, Guangzhou',
      email: 'ed.hellen@idesign.com'
    },
    {
      key: '5',
      id: 647616735,
      name: 'William Smith',
      age: 25,
      salary: 27000,
      address: '55 Park Road, Guangzhou',
      email: 'william.smith@idesign.com'
    },
    {
      key: '6',
      id: 616736475,
      name: 'Tim Cook',
      age: 26,
      salary: 36000,
      address: '66 Park Road, Guangzhou',
      email: 'tim.cook@idesign.com'
    },
    {
      key: '7',
      id: 475761636,
      name: 'Bill Gates',
      age: 27,
      salary: 12000,
      address: '77 Park Road, Guangzhou',
      email: 'bill.gates@idesign.com'
    },
    {
      key: '8',
      id: 364616775,
      name: 'Steve Jobs',
      age: 28,
      salary: 66000,
      address: '88 Park Road, Guangzhou',
      email: 'steve.jobs@idesign.com'
    }
  ];

  return (
    <Table columns={columns} data={data} />
  );
};

export default App;
```

## 设最大高度

通过 `height` 属性指定表格内容最大高度，超出显示滚动条：

```tsx
import React from 'react';
import { Table } from 'idesign-react';

const App = () => {
  const columns = [
    {
      title: 'ID',
      key: 'id'
    },
    {
      title: 'Name',
      key: 'name'
    },
    {
      title: 'Age',
      key: 'age'
    },
    {
      title: 'Salary',
      key: 'salary'
    },
    {
      title: 'Address',
      key: 'address'
    },
    {
      title: 'Email',
      key: 'email'
    }
  ];

  const data = [
    {
      key: '1',
      id: 103761657,
      name: 'Jane Doe',
      age: 21,
      salary: 23000,
      address: '11 Park Road, Guangzhou',
      email: 'jane.doe@idesign.com'
    },
    {
      key: '2',
      id: 761651037,
      name: 'Alisa Ross',
      age: 22,
      salary: 25000,
      address: '22 Park Road, Guangzhou',
      email: 'alisa.ross@idesign.com'
    },
    {
      key: '3',
      id: 373567616,
      name: 'Kevin Sandra',
      age: 23,
      salary: 22000,
      address: '33 Park Road, Guangzhou',
      email: 'kevin.sandra@idesign.com'
    },
    {
      key: '4',
      id: 735646761,
      name: 'Ed Hellen',
      age: 24,
      salary: 17000,
      address: '44 Park Road, Guangzhou',
      email: 'ed.hellen@idesign.com'
    },
    {
      key: '5',
      id: 647616735,
      name: 'William Smith',
      age: 25,
      salary: 27000,
      address: '55 Park Road, Guangzhou',
      email: 'william.smith@idesign.com'
    },
    {
      key: '6',
      id: 616736475,
      name: 'Tim Cook',
      age: 26,
      salary: 36000,
      address: '66 Park Road, Guangzhou',
      email: 'tim.cook@idesign.com'
    },
    {
      key: '7',
      id: 475761636,
      name: 'Bill Gates',
      age: 27,
      salary: 12000,
      address: '77 Park Road, Guangzhou',
      email: 'bill.gates@idesign.com'
    },
    {
      key: '8',
      id: 364616775,
      name: 'Steve Jobs',
      age: 28,
      salary: 66000,
      address: '88 Park Road, Guangzhou',
      email: 'steve.jobs@idesign.com'
    },
    {
      key: '9',
      id: 103761657,
      name: 'Jane Doe',
      age: 21,
      salary: 23000,
      address: '11 Park Road, Guangzhou',
      email: 'jane.doe@idesign.com'
    },
    {
      key: '10',
      id: 761651037,
      name: 'Alisa Ross',
      age: 22,
      salary: 25000,
      address: '22 Park Road, Guangzhou',
      email: 'alisa.ross@idesign.com'
    },
    {
      key: '11',
      id: 373567616,
      name: 'Kevin Sandra',
      age: 23,
      salary: 22000,
      address: '33 Park Road, Guangzhou',
      email: 'kevin.sandra@idesign.com'
    },
    {
      key: '12',
      id: 735646761,
      name: 'Ed Hellen',
      age: 24,
      salary: 17000,
      address: '44 Park Road, Guangzhou',
      email: 'ed.hellen@idesign.com'
    },
    {
      key: '13',
      id: 647616735,
      name: 'William Smith',
      age: 25,
      salary: 27000,
      address: '55 Park Road, Guangzhou',
      email: 'william.smith@idesign.com'
    },
    {
      key: '14',
      id: 616736475,
      name: 'Tim Cook',
      age: 26,
      salary: 36000,
      address: '66 Park Road, Guangzhou',
      email: 'tim.cook@idesign.com'
    },
    {
      key: '15',
      id: 475761636,
      name: 'Bill Gates',
      age: 27,
      salary: 12000,
      address: '77 Park Road, Guangzhou',
      email: 'bill.gates@idesign.com'
    },
    {
      key: '16',
      id: 364616775,
      name: 'Steve Jobs',
      age: 28,
      salary: 66000,
      address: '88 Park Road, Guangzhou',
      email: 'steve.jobs@idesign.com'
    }
  ];

  return (
    <Table maxHeight={300} columns={columns} data={data} />
  );
};

export default App;
```

## 条纹表格

通过 `stripe` 属性指定表格双数行带条纹：

```tsx
import React from 'react';
import { Table } from 'idesign-react';

const App = () => {
  const columns = [
    {
      title: 'ID',
      key: 'id'
    },
    {
      title: 'Name',
      key: 'name'
    },
    {
      title: 'Age',
      key: 'age'
    },
    {
      title: 'Salary',
      key: 'salary'
    },
    {
      title: 'Address',
      key: 'address'
    },
    {
      title: 'Email',
      key: 'email'
    }
  ];

  const data = [
    {
      key: '1',
      id: 103761657,
      name: 'Jane Doe',
      age: 21,
      salary: 23000,
      address: '11 Park Road, Guangzhou',
      email: 'jane.doe@idesign.com'
    },
    {
      key: '2',
      id: 761651037,
      name: 'Alisa Ross',
      age: 22,
      salary: 25000,
      address: '22 Park Road, Guangzhou',
      email: 'alisa.ross@idesign.com'
    },
    {
      key: '3',
      id: 373567616,
      name: 'Kevin Sandra',
      age: 23,
      salary: 22000,
      address: '33 Park Road, Guangzhou',
      email: 'kevin.sandra@idesign.com'
    },
    {
      key: '4',
      id: 735646761,
      name: 'Ed Hellen',
      age: 24,
      salary: 17000,
      address: '44 Park Road, Guangzhou',
      email: 'ed.hellen@idesign.com'
    },
    {
      key: '5',
      id: 647616735,
      name: 'William Smith',
      age: 25,
      salary: 27000,
      address: '55 Park Road, Guangzhou',
      email: 'william.smith@idesign.com'
    },
    {
      key: '6',
      id: 616736475,
      name: 'Tim Cook',
      age: 26,
      salary: 36000,
      address: '66 Park Road, Guangzhou',
      email: 'tim.cook@idesign.com'
    },
    {
      key: '7',
      id: 475761636,
      name: 'Bill Gates',
      age: 27,
      salary: 12000,
      address: '77 Park Road, Guangzhou',
      email: 'bill.gates@idesign.com'
    },
    {
      key: '8',
      id: 364616775,
      name: 'Steve Jobs',
      age: 28,
      salary: 66000,
      address: '88 Park Road, Guangzhou',
      email: 'steve.jobs@idesign.com'
    }
  ];

  return (
    <Table stripe columns={columns} data={data} />
  );
};

export default App;
```

<API />
