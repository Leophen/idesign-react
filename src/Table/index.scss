.i-table {
  width: 100%;
  color: var(--i-font-2);
  font-size: 14px;
}

.i-table-thead__wrapper,
.i-table-tbody__wrapper {
  display: table;
  width: 100%;
  min-width: 100%;
  margin: 0;
  table-layout: fixed;
  border-collapse: separate; // 合并相邻单元格边框
  border-spacing: 0;
}

.i-table-tbody__box {
  box-sizing: border-box;
  overflow: auto;
  overflow: overlay;
  border: 1px solid var(--i-border);
  border-radius: 0 0 8px 8px;
  &::-webkit-scrollbar-button {
    display: none;
  }
}

.i-table-thead {
  background: var(--i-bg-hover);
  .i-table-th {
    border-top: 1px solid var(--i-border);
    &:first-of-type {
      border-left: 1px solid var(--i-border);
      border-radius: 8px 0 0 0;
    }
    &:last-of-type {
      border-right: 1px solid var(--i-border);
      border-radius: 0 8px 0 0;
    }
  }
}

.i-table-th,
.i-table-td {
  box-sizing: border-box;
  padding: 9px 16px;
  text-align: left;
  border: unset;
  &:first-of-type {
    border-left: none;
  }
  &:last-of-type {
    border-right: none;
  }
}

.i-table-td {
  border-bottom: 1px solid var(--i-border);
}

.i-table-tbody {
  overflow: auto;
  overflow: overlay;
  background: var(--i-bg);
  .i-table-tr {
    background: unset;
    &:last-of-type {
      .i-table-td {
        border-bottom: none;
        &:first-of-type {
          border-radius: 0 0 0 8px;
        }
        &:last-of-type {
          border-radius: 0 0 8px 0;
        }
      }
    }
  }
  &.i-table-tbody__stripe {
    .i-table-tr:nth-of-type(even) {
      background: var(--i-bg-hover);
    }
  }
}
