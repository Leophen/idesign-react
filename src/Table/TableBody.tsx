import React from 'react';
import classNames from 'classnames';
import './index.scss';
import ColumnGroup from './ColumnGroup';
import { TableBodyProps } from './type';

const TableBody: React.FC<TableBodyProps> = (props) => {
  const {
    data = [],
    columns = [],
    maxHeight,
    stripe = false
  } = props

  return (
    <div
      className="i-table-tbody__box"
      style={{ maxHeight }}
    >
      <table className="i-table-tbody__wrapper">
        <ColumnGroup columns={columns} />
        <tbody
          className={classNames(
            'i-table-tbody',
            stripe && 'i-table-tbody__stripe'
          )}
        >
          {data.map((item) => (
            <tr
              className='i-table-tr'
              key={item.key}
            >
              {columns.map((cell) => (
                <td className="i-table-td" key={cell.key}>
                  {item[cell.key]}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default TableBody;
