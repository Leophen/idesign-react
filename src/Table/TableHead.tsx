import React from 'react';
import './index.scss';
import ColumnGroup from './ColumnGroup';
import { TableHeadProps } from './type';

const TableHead: React.FC<TableHeadProps> = (props) => {
  const {
    columns = []
  } = props

  return (
    <table className="i-table-thead__wrapper">
      <ColumnGroup columns={columns} />
      <thead className='i-table-thead'>
        <tr className='i-table-tr'>
          {columns.map((column) => (
            <th className="i-table-th" key={column.key}>
              {column.title}
            </th>
          ))}
        </tr>
      </thead>
    </table>
  );
};

export default TableHead;
