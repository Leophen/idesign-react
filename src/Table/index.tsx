import React from 'react';
import classNames from 'classnames';
import './index.scss';
import TableHead from './TableHead';
import TableBody from './TableBody';
import { TableProps } from './type';

const Table: React.FC<TableProps> = (props) => {
  const {
    className,
    style,
    columns = [],
    data = [],
    maxHeight,
    stripe = false,
    ...restProps
  } = props;

  return (
    <div
      className={classNames(
        'i-table',
        className
      )}
      style={{ ...style }}
      {...restProps}
    >
      <TableHead columns={columns}></TableHead>
      <TableBody
        columns={columns}
        data={data}
        maxHeight={maxHeight}
        stripe={stripe}
      />
    </div>
  )
};

Table.displayName = 'Table';

export default Table;
