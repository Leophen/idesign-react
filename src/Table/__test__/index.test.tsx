import { render } from '@testing-library/react';
import Table from '../index';

const columns = [
  {
    title: 'ID',
    key: 'id'
  },
  {
    title: 'Name',
    key: 'name'
  },
  {
    title: 'Age',
    key: 'age'
  },
  {
    title: 'Salary',
    key: 'salary'
  },
  {
    title: 'Address',
    key: 'address'
  },
  {
    title: 'Email',
    key: 'email'
  }
];

const data = [
  {
    key: '1',
    id: 103761657,
    name: 'Jane Doe',
    age: 21,
    salary: 23000,
    address: '11 Park Road, Guangzhou',
    email: 'jane.doe@idesign.com'
  },
  {
    key: '2',
    id: 761651037,
    name: 'Alisa Ross',
    age: 22,
    salary: 25000,
    address: '22 Park Road, Guangzhou',
    email: 'alisa.ross@idesign.com'
  },
  {
    key: '3',
    id: 373567616,
    name: 'Kevin Sandra',
    age: 23,
    salary: 22000,
    address: '33 Park Road, Guangzhou',
    email: 'kevin.sandra@idesign.com'
  },
  {
    key: '4',
    id: 735646761,
    name: 'Ed Hellen',
    age: 24,
    salary: 17000,
    address: '44 Park Road, Guangzhou',
    email: 'ed.hellen@idesign.com'
  },
  {
    key: '5',
    id: 647616735,
    name: 'William Smith',
    age: 25,
    salary: 27000,
    address: '55 Park Road, Guangzhou',
    email: 'william.smith@idesign.com'
  },
  {
    key: '6',
    id: 616736475,
    name: 'Tim Cook',
    age: 26,
    salary: 36000,
    address: '66 Park Road, Guangzhou',
    email: 'tim.cook@idesign.com'
  },
  {
    key: '7',
    id: 475761636,
    name: 'Bill Gates',
    age: 27,
    salary: 12000,
    address: '77 Park Road, Guangzhou',
    email: 'bill.gates@idesign.com'
  },
  {
    key: '8',
    id: 364616775,
    name: 'Steve Jobs',
    age: 28,
    salary: 66000,
    address: '88 Park Road, Guangzhou',
    email: 'steve.jobs@idesign.com'
  }
];

describe('Table 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Table columns={columns} data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

