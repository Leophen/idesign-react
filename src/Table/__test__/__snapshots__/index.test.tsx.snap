// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-table"
  >
    <table
      class="i-table-thead__wrapper"
    >
      <colgroup />
      <thead
        class="i-table-thead"
      >
        <tr
          class="i-table-tr"
        >
          <th
            class="i-table-th"
          >
            ID
          </th>
          <th
            class="i-table-th"
          >
            Name
          </th>
          <th
            class="i-table-th"
          >
            Age
          </th>
          <th
            class="i-table-th"
          >
            Salary
          </th>
          <th
            class="i-table-th"
          >
            Address
          </th>
          <th
            class="i-table-th"
          >
            Email
          </th>
        </tr>
      </thead>
    </table>
    <div
      class="i-table-tbody__box"
    >
      <table
        class="i-table-tbody__wrapper"
      >
        <colgroup />
        <tbody
          class="i-table-tbody"
        >
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              103761657
            </td>
            <td
              class="i-table-td"
            >
              Jane Doe
            </td>
            <td
              class="i-table-td"
            >
              21
            </td>
            <td
              class="i-table-td"
            >
              23000
            </td>
            <td
              class="i-table-td"
            >
              11 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              jane.doe@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              761651037
            </td>
            <td
              class="i-table-td"
            >
              Alisa Ross
            </td>
            <td
              class="i-table-td"
            >
              22
            </td>
            <td
              class="i-table-td"
            >
              25000
            </td>
            <td
              class="i-table-td"
            >
              22 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              alisa.ross@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              373567616
            </td>
            <td
              class="i-table-td"
            >
              Kevin Sandra
            </td>
            <td
              class="i-table-td"
            >
              23
            </td>
            <td
              class="i-table-td"
            >
              22000
            </td>
            <td
              class="i-table-td"
            >
              33 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              kevin.sandra@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              735646761
            </td>
            <td
              class="i-table-td"
            >
              Ed Hellen
            </td>
            <td
              class="i-table-td"
            >
              24
            </td>
            <td
              class="i-table-td"
            >
              17000
            </td>
            <td
              class="i-table-td"
            >
              44 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              ed.hellen@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              647616735
            </td>
            <td
              class="i-table-td"
            >
              William Smith
            </td>
            <td
              class="i-table-td"
            >
              25
            </td>
            <td
              class="i-table-td"
            >
              27000
            </td>
            <td
              class="i-table-td"
            >
              55 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              william.smith@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              616736475
            </td>
            <td
              class="i-table-td"
            >
              Tim Cook
            </td>
            <td
              class="i-table-td"
            >
              26
            </td>
            <td
              class="i-table-td"
            >
              36000
            </td>
            <td
              class="i-table-td"
            >
              66 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              tim.cook@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              475761636
            </td>
            <td
              class="i-table-td"
            >
              Bill Gates
            </td>
            <td
              class="i-table-td"
            >
              27
            </td>
            <td
              class="i-table-td"
            >
              12000
            </td>
            <td
              class="i-table-td"
            >
              77 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              bill.gates@idesign.com
            </td>
          </tr>
          <tr
            class="i-table-tr"
          >
            <td
              class="i-table-td"
            >
              364616775
            </td>
            <td
              class="i-table-td"
            >
              Steve Jobs
            </td>
            <td
              class="i-table-td"
            >
              28
            </td>
            <td
              class="i-table-td"
            >
              66000
            </td>
            <td
              class="i-table-td"
            >
              88 Park Road, Guangzhou
            </td>
            <td
              class="i-table-td"
            >
              steve.jobs@idesign.com
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
